backend/service/github/github.go

```diff
@@ -56,6 +56,23 @@ type RemoteRef struct {
 	Ref string
 }
 
+// BranchRef git reference object representing a "branch"
+type BranchRef struct {
+	Name string
+}
+
+// RepositoryOwner represents the entity that owns a repo
+type RepositoryOwner struct {
+	Login string
+}
+
+// Repository contains information about a requested repository.
+type Repository struct {
+	Name          string
+	Owner         RepositoryOwner
+	DefaultBranch BranchRef
```

Comment:
nit: flatten to string, we don't need to adhere to graphQL abstractions in our internal objects